File: src/main/GO/leetcode/lists/topinterview/LeetCode88.go
package topinterview

func merge(nums1 []int, m int, nums2 []int, n int) {
	index := m + n - 1
	i, j := m-1, n-1
	for i >= 0 || j >= 0 {
		if i >= 0 && j >= 0 {
			if nums1[i] < nums2[j] {
				nums1[index] = nums2[j]
				j--
				index--
			} else {
				nums1[index] = nums1[i]
				i--
				index--
			}
		} else if i >= 0 {
			nums1[index] = nums1[i]
			i--
			index--
		} else if j >= 0 {
			nums1[index] = nums2[j]
			j--
			index--
		}
	}
}

File: src/main/java/leetcode/plan/datastructure/level1/LeetCode88.java
package leetcode.plan.datastructure.level1;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/10/6 18:08
 */
public class LeetCode88 {

    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int index1 = m + n - 1;
        int a = m - 1;
        int b = n - 1;

        while (a >= 0 && b >= 0) {
            if (nums1[a] < nums2[b]) {
                nums1[index1] = nums2[b];
                b--;
            } else {
                nums1[index1] = nums1[a];
                a--;
            }
            index1--;
        }
        while (a >= 0) {
            nums1[index1] = nums1[a];
            a--;
            index1--;
        }
        while (b >= 0) {
            nums1[index1] = nums2[b];
            b--;
            index1--;
        }

    }
}
